docs/root/intro/arch_overview/http/http_routing.rst

```diff
@@ -295,3 +295,74 @@ The only inputs supported are request headers (via :ref:`HttpRequestHeaderMatchI
 
 .. tip::
    See the docs for the :ref:`matching API <arch_overview_matching_api>` for more information about the API as a whole.
+
+Sublinear Route Matching
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+An incoming request to Envoy needs to be matched to a cluster based on defined `routes <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-virtualhost>`_. Typically, a well understood, linear route search matching with O(n) search cost (see  `Virtual Host <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-virtualhost>`_ → `Routes <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-route>`_) is employed which can become a scalability issue with higher latencies as the number of routes go up to o(1k+). To overcome these scalability challenges the Generic Matcher API ( `Virtual Host → matcher{} <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-virtualhost>`_) → `matcher_tree <https://www.envoyproxy.io/docs/envoy/latest/xds/type/matcher/v3/matcher.proto#envoy-v3-api-msg-xds-type-matcher-v3-matcher-matchertree>`_ can offer a robust and flexible framework for route matching with two distinct sublinear matching implementations:
```

Comment:
```suggestion
An incoming request to Envoy needs to be matched to a cluster based on defined :ref:`routes <envoy_v3_api_msg_config.route.v3.VirtualHost>`_. Typically, a well understood, linear route search matching with O(n) search cost (see  :ref:`Virtual Host <envoy_v3_api_msg_config.route.v3.VirtualHost>`_ → :ref:`Routes <envoy_v3_api_msg_config.route.v3.Route>`_) is employed which can become a scalability issue with higher latencies as the number of routes go up to o(1k+). To overcome these scalability challenges the Generic Matcher API ( `Virtual Host → matcher{} <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-virtualhost>`_) → `matcher_tree <https://www.envoyproxy.io/docs/envoy/latest/xds/type/matcher/v3/matcher.proto#envoy-v3-api-msg-xds-type-matcher-v3-matcher-matchertree>`_ can offer a robust and flexible framework for route matching with two distinct sublinear matching implementations:
```

internal links need to use `:ref:` - ive updated a couple, but im not 100% that ive pointed the links to the correct place

you can find the ref target by clicking "View page source" on the rendered docs pages

not sure exactly the issue that is preventing the docs from building - but its related to these links

Reply:
@phlax all builds passing now - PTAL